// Code generated by smithy-go-codegen DO NOT EDIT.

package types

import (
	"fmt"
	smithy "github.com/awslabs/smithy-go"
	"github.com/awslabs/smithy-go/ptr"
)

// The request was rejected because the requester does not have permission to
// perform the requested operation.
type CloudHsmAccessDeniedException struct {
	Message *string
}

func (e *CloudHsmAccessDeniedException) Error() string {
	return fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())
}
func (e *CloudHsmAccessDeniedException) ErrorMessage() string {
	if e.Message == nil {
		return ""
	}
	return *e.Message
}
func (e *CloudHsmAccessDeniedException) ErrorCode() string             { return "CloudHsmAccessDeniedException" }
func (e *CloudHsmAccessDeniedException) ErrorFault() smithy.ErrorFault { return smithy.FaultClient }
func (e *CloudHsmAccessDeniedException) GetMessage() string {
	return ptr.ToString(e.Message)
}
func (e *CloudHsmAccessDeniedException) HasMessage() bool {
	return e.Message != nil
}

// The request was rejected because of an AWS CloudHSM internal failure. The
// request can be retried.
type CloudHsmInternalFailureException struct {
	Message *string
}

func (e *CloudHsmInternalFailureException) Error() string {
	return fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())
}
func (e *CloudHsmInternalFailureException) ErrorMessage() string {
	if e.Message == nil {
		return ""
	}
	return *e.Message
}
func (e *CloudHsmInternalFailureException) ErrorCode() string {
	return "CloudHsmInternalFailureException"
}
func (e *CloudHsmInternalFailureException) ErrorFault() smithy.ErrorFault { return smithy.FaultServer }
func (e *CloudHsmInternalFailureException) GetMessage() string {
	return ptr.ToString(e.Message)
}
func (e *CloudHsmInternalFailureException) HasMessage() bool {
	return e.Message != nil
}

// The request was rejected because it is not a valid request.
type CloudHsmInvalidRequestException struct {
	Message *string
}

func (e *CloudHsmInvalidRequestException) Error() string {
	return fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())
}
func (e *CloudHsmInvalidRequestException) ErrorMessage() string {
	if e.Message == nil {
		return ""
	}
	return *e.Message
}
func (e *CloudHsmInvalidRequestException) ErrorCode() string {
	return "CloudHsmInvalidRequestException"
}
func (e *CloudHsmInvalidRequestException) ErrorFault() smithy.ErrorFault { return smithy.FaultClient }
func (e *CloudHsmInvalidRequestException) GetMessage() string {
	return ptr.ToString(e.Message)
}
func (e *CloudHsmInvalidRequestException) HasMessage() bool {
	return e.Message != nil
}

// The request was rejected because it refers to a resource that cannot be found.
type CloudHsmResourceNotFoundException struct {
	Message *string
}

func (e *CloudHsmResourceNotFoundException) Error() string {
	return fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())
}
func (e *CloudHsmResourceNotFoundException) ErrorMessage() string {
	if e.Message == nil {
		return ""
	}
	return *e.Message
}
func (e *CloudHsmResourceNotFoundException) ErrorCode() string {
	return "CloudHsmResourceNotFoundException"
}
func (e *CloudHsmResourceNotFoundException) ErrorFault() smithy.ErrorFault { return smithy.FaultClient }
func (e *CloudHsmResourceNotFoundException) GetMessage() string {
	return ptr.ToString(e.Message)
}
func (e *CloudHsmResourceNotFoundException) HasMessage() bool {
	return e.Message != nil
}

// The request was rejected because an error occurred.
type CloudHsmServiceException struct {
	Message *string
}

func (e *CloudHsmServiceException) Error() string {
	return fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())
}
func (e *CloudHsmServiceException) ErrorMessage() string {
	if e.Message == nil {
		return ""
	}
	return *e.Message
}
func (e *CloudHsmServiceException) ErrorCode() string             { return "CloudHsmServiceException" }
func (e *CloudHsmServiceException) ErrorFault() smithy.ErrorFault { return smithy.FaultClient }
func (e *CloudHsmServiceException) GetMessage() string {
	return ptr.ToString(e.Message)
}
func (e *CloudHsmServiceException) HasMessage() bool {
	return e.Message != nil
}

// The request was rejected because of a tagging failure. Verify the tag conditions
// in all applicable policies, and then retry the request.
type CloudHsmTagException struct {
	Message *string
}

func (e *CloudHsmTagException) Error() string {
	return fmt.Sprintf("%s: %s", e.ErrorCode(), e.ErrorMessage())
}
func (e *CloudHsmTagException) ErrorMessage() string {
	if e.Message == nil {
		return ""
	}
	return *e.Message
}
func (e *CloudHsmTagException) ErrorCode() string             { return "CloudHsmTagException" }
func (e *CloudHsmTagException) ErrorFault() smithy.ErrorFault { return smithy.FaultClient }
func (e *CloudHsmTagException) GetMessage() string {
	return ptr.ToString(e.Message)
}
func (e *CloudHsmTagException) HasMessage() bool {
	return e.Message != nil
}
